cmd/root.go

```diff
@@ -152,3 +154,32 @@ func getHome() (string, error) {
 
 	return "", errors.New("Can't get home directory")
 }
+
+// generateRandomBytes returns securely generated random bytes.
+// It will return an error if the system's secure random
+// number generator fails to function correctly, in which
+// case the caller should not continue.
+func generateRandomBytes(n int) ([]byte, error) {
+	b := make([]byte, n)
+	_, err := rand.Read(b)
+	// Note that err == nil only if we read len(b) bytes.
+	if err != nil {
+		return nil, err
+	}
+
+	return b, nil
+}
+
+// generateEncodedRandomPassword returns a standard base64 encoded
+// securely generated random string.
+// It will return an error if the system's secure random
+// number generator fails to function correctly, in which
+// case the caller should not continue.
+func generateEncodedRandomPassword(s int) (string, error) {
+	b, err := generateRandomBytes(s)
+	if err != nil {
+		return "", err
+	}
+	pw := base64.StdEncoding.EncodeToString(b)
```

Comment:
are you encoding twice?

Reply:
yes. See this https://github.com/kubeapps/kubeapps/pull/51#discussion_r152884529